src/main/java/decodes/cwms/CwmsTimeSeriesDb.java

```diff
@@ -859,6 +859,9 @@ public DbKey connect( String appName, Properties credentials )
 				+ " with officeID=" + dbOfficeId);
 
 				postConnectInit(appName, conn); // Make sure the versions and such are set
+				setConnection(conn);
+				setupKeyGenerator(); // TODO: this should take the connection as a parameter
+				setConnection(null);
```

Comment:
should this be wrapped in a `finally` block in case `setupKeyGenerator()` fails?